### PR TITLE
feat: add depositTransaction trait

### PR DESCRIPTION
**Pull request by hoank101**
This PR will help this issue https://github.com/paradigmxyz/reth/issues/11736
 converted into a trait from reth to alloy-rs

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/consensus/src/transaction/deposit.rs [hunk 1]
```diff
@@ -241,6 +241,31 @@ impl Decodable for TxDeposit {
     }
 }
 
+pub trait DepositTransaction {
+    fn source_hash(&self) -> Option<B256>;
+    fn mint(&self) -> Option<u128>;
+    fn is_system_transaction(&self) -> bool;
+    fn is_deposit(&self) -> bool;
+}
```
Comment (mattsse) [inline crates/consensus/src/transaction/deposit.rs:249]: this needs some docs

Comment (mattsse) [inline crates/consensus/src/transaction/deposit.rs:249]: I'd also like to move this to a new file `src/traits.rs`

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/consensus/src/transaction/deposit.rs (hunk 1 above)
Comment (hoank101) [reply crates/consensus/src/transaction/deposit.rs]: thank @mattsse, j just update, please help me review it!

**Review by emhane — APPROVED** [review]

**Review by refcell — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
@refcell context is here that we need this as prep for moving reth off of optimism features

**Pull request merged**